config/config.go

```diff
@@ -849,6 +853,19 @@ func (c *ScrapeConfig) Validate(globalConfig GlobalConfig) error {
 		c.MetricNameValidationScheme = globalConfig.MetricNameValidationScheme
 	}
 
+	switch globalConfig.MetricNameEscapingScheme {
+	case model.AllowUTF8:
+		if c.MetricNameValidationScheme != UTF8ValidationConfig {
+			return errors.New("utf8 metric names requested but validation scheme is not set to UTF8")
+		}
+	case "", model.EscapeUnderscores, model.EscapeDots, model.EscapeValues:
```

Comment:
What `""` means (how user can find this info)? Shouldn't we have a default for both escape and validation in https://github.com/prometheus/prometheus/blob/95f49dd84be7f2ef5471f6ce5cf448b6a14d8598/config/config.go#L163 for brevity?


Alternatively we can mention in docs what empty means.

Reply:
Yea, looking ahead looks like empty validation means "UTF8ValidationConfig", empty schema switches defaults depending on validation. Then I would suggest:

1. Adding DefaultGlobalConfig.MetricNameValidationScheme = UTF8ValidationConfig
2. Resolving what you resolve in complex `func (sp *scrapePool) schemesFromConfig() (validationScheme model.ValidationScheme, escapingScheme model.EscapingScheme, err error) {` function here in the config layer. 

The higher/earlier we do complex things, the better. Also we have opportunity to have this complexity in a single place to debug/maintain vs 2 places.